util/efd_sched.py

```diff
@@ -0,0 +1,231 @@
+    """Load a scheduler, conditions tuple from a URI and optionally save it.
+
+    Parameters
+    ----------
+    scheduler_url : `str`
+        The URL of the scheduler to load.
+    destination : `str`, optional
+        The file path to save the scheduler to.
+        If None, the (scheduler, conditions) tuple will be returned.
+
+    Returns
+    -------
+    result : `tuple` or `str`
+        If `destination` is None, the loaded scheduler (scheduler, conditions)
+        tuple is returned.
+        Otherwise, the file path of the saved scheduler is returned.
+
+    Notes
+    -----
+    This function sets the environment variable
+    `LSST_DISABLE_BUCKET_VALIDATION` to "1".
+    """
+
+    os.environ["LSST_DISABLE_BUCKET_VALIDATION"] = "1"
+    scheduler_resource_path = ResourcePath(scheduler_url.replace("https://s3.cp.lsst.org/", "s3://rubin:"))
+    scheduler_pickle_bytes = scheduler_resource_path.read()
+
+    if destination is None:
+        result = pickle.loads(scheduler_pickle_bytes)
+    else:
+        if os.path.isdir(destination):
+            base_file_name = (
+                scheduler_resource_path.basename()
+                .replace("Scheduler:2_Scheduler:2", "auxtel")
+                .replace(":", "")
+                + ".xz"
+            )
+            file_name = os.path.join(destination, base_file_name)
+        else:
+            file_name = destination
+
+        this_open = lzma.open if file_name.endswith(".xz") else open
+        with this_open(file_name, "wb") as this_file:
+            this_file.write(scheduler_pickle_bytes)
+
+        result = file_name
+
+    return result
+
+
+async def get_scheduler_at_time(desired_time, destination=None, efd="usdf_efd"):
+    """Get the scheduler, conditions tuple for a desired time.
+
+    Parameters
+    ----------
+    desired_time : `str`
+        The desired time to get the scheduler for in iso8601 format.
+    destination : `str`, optional
+        The file path to save the scheduler to.
+        If None, the (scheduler, conditions) tuple is returned.
+    efd : `str`, optional
+        The EFD to query for schedulers. Defaults to 'usdf_efd'.
+
+    Returns
+    -------
+    result : `tuple` or `str`
+        If `destination` is None, the loaded (scheduler, conditions) tuple
+        is returned.
+        Otherwise, the file path of the saved scheduler is returned.
+    """
+    these_scheduler_references = await query_schedulers_in_window(desired_time, efd)
+    this_url = these_scheduler_references.url[0]
+    result = get_scheduler(this_url, destination)
+    return result
+
+
+async def get_scheduler_on_night(night, scheduler_index, destination, efd="usdf_efd"):
+    """Get a scheduler for a given night from the EFD.
+
+    Parameters
+    ----------
+    night : `str`
+        The night for which to get the scheduler.
+    scheduler_index : `int`, optional
+        The index of the scheduler to retrieve (default is 0).
```

Comment:
I suspect that "index" is an overloaded term, when there is also salIndex running around. 
In general, I'd suggest this does not need to exist as-is .. people can get the list of all the schedulers in a night above, then pick out the one they want to load, and then load it. 
This being a simpler thing of "get the latest scheduler" would make sense though. 

Note that "these_scheduler_references.url[index]" is triggering warnings in the most recent version of pandas and probably needs to swap to these_scheduler_references.url.iloc[index].

Reply:
I'm mostly trying to avoid doing extra typing or cutting-and-pasting by letting the user pick a small integer instead. How about if I replace the word `index` with `row_number`?